pkg/kdump/config-client-suse.js

```diff
@@ -0,0 +1,262 @@
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { ConfigFile } from './config-client.js';
+
+/* Parse an dotenv-style config file
+ * and monitor it for changes
+ */
+export class ConfigFileSUSE extends ConfigFile {
+    /* parse lines of the config file
+     * if a line has a valid key=value format, use the key in _internal structure
+     * and also store original line, line index, value and optional line suffix / comment
+     * if value was quoted it will be stripped of quotes in `value` and `quoted` flag will
+     * be used when writing the file to keep original formatting
+     * e.g. for line 'someKey="foo" # comment'
+     * outputObject._internal["someKey"] = {
+     *      index: 0,
+     *      value: "foo",
+     *      quoted: true,
+     *      origLine: "someKey=\"foo\" # comment",
+     *      suffix: "# comment"
+     * }
+     * skipNotify: Don't notify about changes, e.g.to avoid multiple updates when writing a file
+     */
+    _parseText(rawContent, skipNotify = false) {
+        this._dataAvailableResolve();
+
+        // clear settings if file is empty/missing
+        if (!rawContent) {
+            this._originalSettings = null;
+            this.settings = null;
+            if (!skipNotify)
+                this.dispatchEvent("kdumpConfigChanged", this.settings);
+            return;
+        }
+
+        // if nothing changed, don't bother parsing the content
+        if (rawContent == this._rawContent)
+            return;
+
+        this._rawContent = rawContent;
+
+        // this is the format expected by the UI
+        this.settings = {
+            _internal: {},
+            targets: {},
+            compression: { enabled: false, allowed: true },
+        };
+
+        this._lines = rawContent.split(/\r?\n/);
+        this._lines.forEach((line, index) => {
+            const trimmed = line.trim();
+            // if the line is empty or only a comment, skip
+            if (trimmed.indexOf("#") === 0 || trimmed.length === 0)
+                return;
+
+            // parse KEY=value or KEY="value" line
+            let parts = trimmed.match(/^([A-Z_]+)=(.*)$/);
+            if (parts === null)
+                return;
+            const key = parts[1];
+            let value = parts[2];
+
+            // value might be quoted
+            let quoted = false;
+            if (value.startsWith("\"")) {
+                quoted = true;
+                parts = value.match(/^"([^"]*)"\s*(.*)$/);
+                // malformed line, no ending quote?
+                if (parts === null) {
+                    console.error("Malformed kdump config line:", line, "in", this.filename);
+                    return;
+                }
+            } else {
+                // not quoted should be simple value
+                parts = value.match(/^(\S+)\s*(.*)$/);
```

Comment:
A line line `key = foo bar` would come out as `settings["key"] == "foo"`, is that really intended? This seems very unintuitive. Wouldn't it make more sense to strip away `#.*$` first, strip the rest, and take that as value?

Or is it important that this file behaves like shell commands? But that would also not simply ignore the second word, that would be a syntax error.

Reply:
Yeah, this should be correct shell variable, but again our tools seem to accept that and fix on write. I will follow this.

Reply:
done